concordium-consensus/src/Concordium/GlobalState/Persistent/Account/StructureV1.hs

```diff
@@ -1340,72 +1343,75 @@ migratePersistentAccount StateMigrationParametersTrivial acc = do
+                        bkrPoolInfo <- refLoad (V0._theExtraBakerInfo _extraBakerInfo)
+                        return $! BakerInfoExV1 bkrInfo bkrPoolInfo
+                    paseBakerInfo' <- refMake bkrInfoEx
+                    (paseBakerInfo, _) <- refFlush paseBakerInfo'
+                    let baker =
+                            PersistentAccountStakeEnduringBaker
+                                { paseBakerRestakeEarnings = _stakeEarnings,
+                                  paseBakerPendingChange = pendingChangeEffectiveTimestamp <$> _bakerPendingChange,
+                                  ..
+                                }
+                    return (_stakedAmount, baker)
+                V0.PersistentAccountStakeDelegate dlgRef -> do
+                    AccountDelegationV1{..} <- lift $ refLoad dlgRef
+                    let del =
+                            PersistentAccountStakeEnduringDelegator
+                                { paseDelegatorRestakeEarnings = _delegationStakeEarnings,
+                                  paseDelegatorId = _delegationIdentity,
+                                  paseDelegatorTarget = _delegationTarget,
+                                  paseDelegatorPendingChange = pendingChangeEffectiveTimestamp <$> _delegationPendingChange
+                                }
+                    return (_delegationStakedAmount, del)
+            paedEncryptedAmount <- do
+                mea <- lift $ do
+                    ea <- refLoad _accountEncryptedAmount
+                    isInit <- isInitialPersistentAccountEncryptedAmount ea
+                    return $ if isInit then Null else Some ea
+                forM mea $ \ea -> do
+                    newEA <- migratePersistentEncryptedAmount ea
+                    refMake $! newEA
+            paedReleaseSchedule <- do
+                mrs <- lift $ do
+                    rs <- refLoad _accountReleaseSchedule
+                    return $ if ARSV0.isEmptyAccountReleaseSchedule rs then Null else Some rs
+                forM mrs $ \rs -> do
+                    newRS <- migrateAccountReleaseScheduleFromV0 rs
+                    rsRef <- refMake $! newRS
+                    return (rsRef, ARSV0.releaseScheduleLockedBalance rs)
+            (accountEnduringData, _) <-
+                refFlush
+                    =<< refMake
+                    =<< makeAccountEnduringData
+                        paedPersistingData
+                        paedEncryptedAmount
+                        paedReleaseSchedule
+                        paedStake
+            return $!
+                PersistentAccount
+                    { accountNonce = _accountNonce,
+                      accountAmount = _accountAmount,
+                      ..
+                    }
+        _ ->
+            error $
+                "migratePersistentAccountFromV0 invariant violation: "
+                    <> "migrationParams should be StateMigrationParametersP4ToP5"
```

Comment:
It was changed because the compiler complained about redundant constraints and I was told to avoid pragmas. Is there a way to do it for a single declaration, rather than the whole file?

Reply:
The pragmas should be avoided, and some should be avoided at all cost, such as disabling completeness checks for pattern matching.

Others are sometimes necessary due to compiler bugs, but they are also not dangerous, such as redundant typeclass constraints. They should still be avoided, but if the choice is between disabling a critical feature or just a good one then that choice should be clear.

I think you have to disable the specific warning per file. I don't know of a way to do it per declaration.